WordPress/wordpress-develop
# Add block-based widget editor

Trac ticket: https://core.trac.wordpress.org/ticket/51506

Supersedes https://github.com/WordPress/wordpress-develop/pull/588.

This moves the the block-based widget editor over from Gutenberg into Core.

Broadly speaking, there's three parts to this:

1. `@wordpress/edit-widgets` is added as a dependency. This is what contains the editor UI.

2. `wp-admin/widgets.php` has been modified to branch between the old form-based editor `wp-admin/widgets-form.php` and the new block-based editor `wp-admin/widgets-block-editor.php` depending on `get_theme_support( 'widgets-block-editor' )` and `use_widgets_block_editor`.

3. Supporting infrastructure such as `WP_Widget_Block` and `widgets.php?widget-preview={}` has been copied over from Gutenberg.

This PR contains `WP_REST_Sidebars_Controller` and `WP_REST_Widget_Utils_Controller` so that it's easier to test, but these parts should be committed separately.

#### How to test

1. `npm run env:start`
2. `npm run env:install`
3. Browse to Appearance → Widgets.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**
